core/primitives/src/views.rs

```diff
@@ -433,6 +434,87 @@ pub struct TrackedShardsView {
+// Used for debug purposes only.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug, Clone)]
+pub struct ApprovalHistoryEntry {
+    // If target_height == base_height + 1  - this is endorsement.
+    // Otherwise this is a skip.
+    pub parent_height: BlockHeight,
+    pub target_height: BlockHeight,
+    // Time when we actually created the approval and sent it out.
+    pub approval_creation_time: DateTime<chrono::Utc>,
+    // The moment when we were ready to send this approval (or skip)
+    pub timer_started_ago_millis: u64,
+    // But we had to wait at least this long before doing it.
+    pub expected_delay_millis: u64, 
+}
+
+// Information about chunk produced by this node.
+// For debug purposes only.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug, Default)]
+pub struct ChunkProduction {
+    // Time when we produced the chunk.
+    pub chunk_production_time: Option<DateTime<chrono::Utc>>,
+    // How long did the chunk production take (reed solomon encoding, preparing fragments etc.)
+    // Doesn't include network latency.
+    pub chunk_production_duration_millis: Option<u64>
+}
+// Information about the block produced by this node.
+// For debug purposes only.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug, Default, Clone)]
+pub struct BlockProduction {
+    // Time at which we received chunk for given shard.
+    pub chunks_collection_time: Vec<Option<DateTime<chrono::Utc>>>, 
+    // Time when we produced the block.
+    pub block_production_time: Option<DateTime<chrono::Utc>> ,
+}
+
+// Information about things related to block/chunk production
+// at given height.
+// For debug purposes only.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug, Default)]
+pub struct ProductionAtHeight {
+    // Approvals that we received.
+    pub approvals: Option<ApprovalAtHeightStatus>,
+    // Block that we produced.
+    pub block_production: Option<BlockProduction>,
+    // Map from shard_id to chunk that we produced for this height.
+    pub chunk_production: HashMap<u64, ChunkProduction>,
+}
+
+// Infromation about the approvals that we received.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug)]
+pub struct ApprovalAtHeightStatus {
+    // Map from validator id to the type of approval that they sent and timestamp.
+    pub approvals: HashMap<String, (ApprovalInner, Option<DateTime<chrono::Utc>>)>,
+    // Time at which we received 2/3 approvals (doomslug threshold).
+    pub ready_at: Option<DateTime<chrono::Utc>>,
+}
+
+// Status of the validator -- used for debug purposes only.
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Debug)]
+pub struct ValidatorStatus {
+    pub validator_name: Option<String>,
```

Comment:
I think every `String` in these newly introduced structs wants to be an `AccountId`